Increase speed

jemorrison opened the pull request:
In this PR I got a little carried away with making pep8 changes. The goal of this PR was some  improvement in the code cube_cloud.py. I took some for loops out.  I plan to make more improvements later to speed this program up but I think it might be getting overwhelming to review these changes so I stopped.

jdavies-st commented:
The PEP8 changes look good!  Which module/function has the speed improvement code?  I'm have a hard time finding it amongst the PEP8 changes.  I do see the 'miripsf' lines.

jemorrison commented:
I should not of made the pep8 changes with the speed improvement changes it got too confusing and
I decided to stop on the speed improvements and delay some of them because I thought it was getting
confusing to  review.
The cube_cloud.py have the first set of speed improvements. I removed some loops and changed it
to array math instead.

nden commented on jwst/cube_build/cube_cloud.py:
```diff
@@ -44,224 +41,241 @@ def match_det2cube(self, input_model,
 
             # transform Cube Spaxel centers to alpha,beta of exposure
-            # for MIRI weighting parameters are based on distance in 
+            # for MIRI weighting parameters are based on distance in
```
I'm not sure I understand the logic for MIRI. `coord1` is overwritten several times. 
From the code section about MIRI I think this is what's coded:
```
- if weighting == 'miripsf':
      -- then coord1 is computed in alpha_beta
          get out of if statement
  else:
     -- if alpha_beta is requested:
           -- coord1 is computed in alpha_beta
        else:
          -- coord1 is computed in ra_dec       
```
If this is the desired logic, then it can be coded like this pseudocode avoiding recomputation of `coord1`.

jemorrison replied:
I want to pull out the miripsf and alpha-beta weighting since they are special cases.
I will do that in a future pr.

nden commented on jwst/cube_build/cube_cloud.py:
```diff
@@ -44,224 +41,241 @@ def match_det2cube(self, input_model,
     spaxel class matched to detector pixels with flux and weighting updated for each
     match
 
-
```
The `Parameters` section does not correspond to the parameter names.

nden commented on jwst/cube_build/cube_cloud.py:
```diff
@@ -44,224 +41,241 @@ def match_det2cube(self, input_model,
     Short Summary
     -------------
-    Get the wavelength normalization weights that we will 
+    Get the wavelength normalization weights that we will
```
This function is currently not used anywhere. If it's here because it will be needed in the future, it may be faster to use a dict instead of a function.

jemorrison replied:
I will remove this. This was an old method that got replaced.

jemorrison commented on jwst/cube_build/cube_cloud.py:
```diff
@@ -44,224 +41,241 @@ def match_det2cube(self, input_model,
+            valid1 = ~np.isnan(coord1)
+#________________________________________________________________________________
+    elif self.instrument == 'NIRSPEC': # for NIRSPEC need to look over every slice
+        # Slice No are provided by ifu_cube.py
```
I want to separate the MIRI weighting methods and pull out when the weighting is based
on the psf then I think it will be cleaner. I decided to wait and just get the pep8 updates done
first.

nden replied:
That's fine.

jemorrison commented:
Well I tried the it once and I could not get the changes made. Did you follow the instructions from 'command line'.

jemorrison commented:
Ok thanks James. That was actually easy. I was trying to make it so much harder than it was.

jemorrison commented:
Ok I have a failure. I have completely messed up my branch of this on my computer.
So is there a way to pull in the version in the PR getting the fixes of the conflicts I just made
and make this fix and also make sure it all looks good.
